pyro/infer/mcmc/mcmc.py

```diff
@@ -244,9 +244,9 @@ class MCMC(TracePosterior):
         CUDA.
     :param bool disable_progbar: Disable progress bar and diagnostics update.
     """
-    def __init__(self, kernel, num_samples, warmup_steps=0,
+    def __init__(self, kernel, num_samples, warmup_steps=None,
                  num_chains=1, mp_context=None, disable_progbar=False):
-        self.warmup_steps = warmup_steps if warmup_steps is not None else num_samples // 2  # Stan
+        self.warmup_steps = num_samples if warmup_steps is None else warmup_steps  # Stan
```

Comment:
Correct the behaviour from Stan: Stan uses `iter` and set default `warmup = iter // 2`. Iter in Stan includes warmup, not separated from warmup. Ref: https://pystan.readthedocs.io/en/latest/api.html#pystan.stan. Because `warmup_steps=None` is a bug previously, this change does not affect any user.